oceannavigator/frontend/src/components/Map.jsx

```diff
@@ -734,6 +753,44 @@ export default class Map extends React.PureComponent {
           this.popupElement.innerHTML = feature.get("name");
         }
         $(this.map.getTarget()).css("cursor", "pointer");
+
+        if (feature.get("type") == "area") {
+          this.map.forEachFeatureAtPixel(e.pixel, function (f) {
+            selected = f;
+            f.setStyle([
+              new olstyle.Style({
+                stroke: new olstyle.Stroke({
+                  color: "#FFFFFF",
+                  width: 2,
+                }),
+                fill: new olstyle.Fill({
+                  color: "#FFFFFF80"
+                })
+              }),
+              new olstyle.Style({
+                stroke: new olstyle.Stroke({
+                  color: "#000000",
+                  width: 1,
+                }),
+              }),                  
+              new olstyle.Style({
+                geometry: new olgeom.Point(olproj.transform(f.get("centroid"), "EPSG:4326", this.props.state.projection)),
+                text: new olstyle.Text({
+                  text: f.get("name"),
+                  font: '14px sans-serif',
+                  fill: new olstyle.Fill({
+                    color: "#000000",
+                  }),
+                  stroke: new olstyle.Stroke({
+                    color: "#FFFFFF", 
+                    width: 2,
+                  }),
+                }),
+              }),  
+            ]);
```

Comment:
It seems like overkill to have to completely redefine the style when I really just want to update the fill for the object. I couldn't find a way to do that in openlayer's documentation. Any suggestions here?

Reply:
No ideas from me. Openlayers seems to have a kind of imperative API where you need to tell it everything all the time.